### PR TITLE
*: add acrpull controller, binding

### PR DESCRIPTION
**Pull request by stevekuznetsov**
### What this PR does

Jira: 
Link to demo recording:

### CODE REVIEW

File: frontend/deploy/helm/frontend/templates/acrpullbinding.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: pull-binding
+spec:
+  acr:
+    environment: PublicCloud
+    scope: {{ .Values.pullBinding.registry }}
+    server: {{ .Values.pullBinding.scope }}
+  auth:
+    workloadIdentity:
+      serviceAccountRef: frontend
```
Comment (stevekuznetsov) [inline frontend/deploy/helm/frontend/templates/acrpullbinding.yaml:12]: The SA whose credentials are projected is not necessarily the same SA as where they are projected, on purpose. So if we wanted a different identity with pull rights federated to some other SA in the namespace, we could.

File: dev-infrastructure/modules/aks-cluster-base.bicep [hunk 2]
```diff
@@ -460,15 +460,26 @@ resource uami_fedcred 'Microsoft.ManagedIdentity/userAssignedIdentities/federate
     parent: uami[i]
     name: '${workloadIdentities[i].value.uamiName}-${location}-fedcred'
     properties: {
-      audiences: [
```
Comment (stevekuznetsov) [inline dev-infrastructure/modules/aks-cluster-base.bicep:463]: @geoberle  please provide feedback on all this

File: frontend/deploy/helm/frontend/templates/acrpullbinding.yaml (hunk 1 above)
Comment (weinong) [reply frontend/deploy/helm/frontend/templates/acrpullbinding.yaml:12]: it's easy to use a separate one now. lets do that.

File: frontend/deploy/helm/frontend/templates/acrpullbinding.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      clientID: {{ .Values.pullBinding.workloadIdentityClientId }}
+      tenantID: {{ .Values.pullBinding.workloadIdentityTenantId }}
```
Comment (stevekuznetsov) [inline frontend/deploy/helm/frontend/templates/acrpullbinding.yaml:14]: Newest version of the controller allows us to reference the puller's MI for this token, letting multiple MI to be federated to the SA

**Pull request closed without merging**